File: tests/fixtures/benchmark_project/frontend/types/product.ts
/**
 * Product-related type definitions.
 */

export enum ProductCategory {
    Electronics = "electronics",
    Clothing = "clothing",
    Food = "food",
    Books = "books",
    Other = "other",
}

export interface Product {
    id: number;
    name: string;
    price: number;
    category: ProductCategory;
    description?: string;
    stockCount: number;
}

export type ProductID = number;

export interface CreateProductRequest {
    name: string;
    price: number;
    category?: ProductCategory;
    description?: string;
}

export function formatPrice(price: number, currency: string = "USD"): string {
    return new Intl.NumberFormat("en-US", {
        style: "currency",
        currency,
    }).format(price);
}

export function isInStock(product: Product): boolean {
    return product.stockCount > 0;
}

File: tests/fixtures/benchmark_project/src/models/product.py
"""Product domain model."""

from dataclasses import dataclass
from enum import Enum


class ProductCategory(Enum):
    ELECTRONICS = "electronics"
    CLOTHING = "clothing"
    FOOD = "food"
    BOOKS = "books"
    OTHER = "other"


@dataclass
class Product:
    id: int
    name: str
    price: float
    category: ProductCategory = ProductCategory.OTHER
    description: str | None = None
    stock_count: int = 0

    def is_available(self) -> bool:
        return self.stock_count > 0

    def apply_discount(self, percentage: float) -> float:
        if not 0 <= percentage <= 100:
            raise ValueError("Discount must be between 0 and 100")
        discount = self.price * (percentage / 100)
        return round(self.price - discount, 2)

    def restock(self, quantity: int) -> None:
        if quantity < 0:
            raise ValueError("Cannot restock negative quantity")
        self.stock_count += quantity

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "name": self.name,
            "price": self.price,
            "category": self.category.value,
            "description": self.description,
            "stock_count": self.stock_count,
        }
